sklearn/metrics/classification.py

```diff
@@ -1101,6 +1217,36 @@ def _prf_divide(numerator, denominator, metric, modifier, average, warn_for):
     return result
 
 
+def _check_set_wise_labels(y_true, y_pred, average, labels, pos_label):
+    """Validation associated with set-wise metrics
+
+    Returns identified labels
+    """
+    average_options = (None, 'micro', 'macro', 'weighted', 'samples')
+    if average not in average_options and average != 'binary':
+        raise ValueError('average has to be one of ' +
+                         str(average_options))
+
+    y_type, y_true, y_pred = _check_targets(y_true, y_pred)
+    present_labels = unique_labels(y_true, y_pred)
+    if average == 'binary':
+        if y_type == 'binary':
+            if pos_label not in present_labels:
+                if len(present_labels) >= 2:
+                    raise ValueError("pos_label=%r is not a valid label: "
+                                     "%r" % (pos_label, present_labels))
+            labels = [pos_label]
+        else:
+            raise ValueError("Target is %s but average='binary'. Please "
+                             "choose another average setting." % y_type)
+    elif pos_label not in (None, 1):
```

Comment:
Maybe we can remove None, pos_label=None is no longer supported and it's ill-defined.

Reply:
I'm not sure this is the right place to change p/r/f/s behaviour, even if legacy.